### DISCUSSION TITLE
Cannot connect to device, but works in gatttool. Public/random type issue?

### DISCUSSION THREAD

**Discussion by matzrh** [hbldh/bleak#1187] [Q&A]
I am trying to connect to a shelly switch from my Rpi4b, which works find through gattool, but the connection is not working with bleak an python.

Code
```
#!/usr/bin/python3

import asyncio
from bleak import BleakScanner
from bleak import BleakClient
devName="ShellyPro2PM"

async def discover():
    devices = await BleakScanner.discover()
    return devices

async def examine(mac):
    async with BleakClient(mac) as client:
        await client.connect()
        print(client.services)

devices=asyncio.run(discover())
try:
   shelly=[x for x in devices if devName in x.name][0]
except IndexError:
   print("{} not found".format(devName))
   exit()
print(shelly.address)
asyncio.run(examine(shelly)) 
```

Output: 
```
EC:62:60:90:E3:B6
Traceback (most recent call last):
  File "./switcher.py", line 32, in <module>
    asyncio.run(examine(shelly))
  File "/usr/lib/python3.7/asyncio/runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.7/asyncio/base_events.py", line 584, in run_until_complete
    return future.result()
  File "./switcher.py", line 19, in examine
    async with BleakClient(mac) as client:
  File "/usr/local/lib/python3.7/dist-packages/bleak/__init__.py", line 433, in __aenter__
    await self.connect()
  File "/usr/local/lib/python3.7/dist-packages/bleak/__init__.py", line 471, in connect
    return await self._backend.connect(**kwargs)
  File "/usr/local/lib/python3.7/dist-packages/bleak/backends/bluezdbus/client.py", line 187, in connect
    f"Device with address {self.address} was not found. It may have been removed from BlueZ when scanning stopped.",
bleak.exc.BleakDeviceNotFoundError: Device with address EC:62:60:90:E3:B6 was not found. It may have been removed from BlueZ when scanning stopped.
```

What I do in gatttool:
`sudo gatttool -i hci0 -b EC:62:60:90:E3:B6 -I -t public `

Then:  connect -> prompt becomes purple, so connection successful
Then: characteristics -> services are listed with handlers and I can "char-write-req" to the appropriate handlers

I cannot connect through gatttool when I use the type 'random'.

I have seen similar issues, here, but the problem seemed to be always something related to the type 'random', which seems to be more common.

**Comment by dlech** [comment]
Same issues as https://github.com/hbldh/bleak/issues/1176?

**Comment by matzrh** [comment]
Yes, I think it is.  I saw that post but the bluetoothctl bit confused me, as I did not know that it can also monitor BLE, so I did not have a closer look.
I will see how can work around, here, there  are some pointers, even if I cannot upgrade to bluez 5.65. Thanks!

**Comment by savvn001** [comment]
Seeing the same issue crop up over and over again. Not sure exactly what causes it, seems to just be random.
Bluez 5.64
